sourcecode/apis/contentauthor/app/Http/Controllers/Admin/LibraryUpgradeController.php

```diff
@@ -176,4 +179,60 @@ public function deleteLibrary(H5PLibrary $library): Response
 
         return response()->noContent();
     }
+
+    public function checkLibrary(H5PLibrary $library): View
+    {
+        $h5pDataFolderName = $library->getLibraryString(true);
+        $disk = Storage::disk();
+        $libsFolder = $disk->path('libraries');
+        $libFolder = $disk->path('libraries/' . $h5pDataFolderName);
+
+        /** @var H5PValidator $validator */
+        $validator = resolve(H5PValidator::class);
+        $libraryData = $validator->getLibraryData($h5pDataFolderName, $libFolder, $libsFolder);
+
+        $editorDep = $library->libraries()
+            ->where('dependency_type', 'editor')
+            ->select('required_library_id')
+            ->pluck('required_library_id');
+        $preloadedDep = $library->libraries()
+            ->where('dependency_type', 'preloaded')
+            ->select('required_library_id')
+            ->pluck('required_library_id');
+
+        if ($libraryData !== false) {
+            foreach (['editorDependencies' => $editorDep, 'preloadedDependencies' => $preloadedDep] as $depType => $deps) {
+                if (array_key_exists($depType, $libraryData)) {
+                    foreach ($libraryData[$depType] as $key => $row) {
+                        $depLib = H5PLibrary::fromMachineName($row['machineName'])
```

Comment:
This data could be retrieved via a join to avoid executing a separate query for each library.